x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_merge_outcome.ts

```diff
@@ -1,26 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-/**
- * Type of result of an automatic three-way merge of three values:
- *   - base version
- *   - current version
- *   - target version
- */
-export enum ThreeWayMergeOutcome {
-  /** Took current version and returned as the merged one. */
-  Current = 'CURRENT',
-
-  /** Took target version and returned as the merged one. */
-  Target = 'TARGET',
-
-  /** Merged three versions successfully into a new one. */
-  Merged = 'MERGED',
```

Comment:
This is probably too bold move. My ask was to remove the `Conflict` value, but not the whole property from the `ThreeWayDiff` object. Please add the `ThreeWayMergeOutcome` back.

Although it's not needed for implementing diff calculations or any UI, it was supposed to be used as a property that might be helpful for debugging and troubleshooting of diff calculations or the whole upgrade workflow in general. I expect it to be especially helpful for understanding the diffing of complex objects, such as `related_integrations` and `threat` arrays.

Reply:
Ok, I'm adding this back. 

Just to be 100% sure I understand your initial idea: with the new `ThreeWayDiffConflict` enum, we don't need the `ThreeWayMergeOutcome.conflict` value anymore, because `ThreeWayMergeOutcome` should only be able to return the following values:

- `Merged`: only for `SOLVABLE` conflicts - a conflict existed between the three version but our algorithm could create a merged solution out of the three
- `Current` and `Target`: two options:
    - There was `NONE` conflict, and the algorithm decided that the returned value should be either `Current` or `Target` (this depends on the algorithm and the specific scenario)
    - There was `NON_SOLVABLE` conflict, no merged version was possible, so we are returning either `Current` or `Target` (this also depends on the algorithm and the specific scenario, but in almost all cases it will be `Current`)

Are we aligned here?

Reply:
@jpdjere Your understanding is exactly right! Thank you.

In other words:

- In trivial cases, such as `AAA`, `ABA`, `AAB`, `ABB`, we return:
  - `ThreeWayDiffConflict`: `NONE`
  - `ThreeWayMergeOutcome`: `Current` or `Target`, depending on the case and algorithm applied to the field
- In the `ABC` cases when it's not possible to auto-merge or we think we should not, we return:
  - `ThreeWayDiffConflict`: `NON_SOLVABLE`
  - `ThreeWayMergeOutcome`: `Current` or `Target`, depending on the algorithm (`Current` in most cases)
- In the `ABC` cases when it's possible to auto-merge and we do it, we return:
  - `ThreeWayDiffConflict`: `SOLVABLE`
  - `ThreeWayMergeOutcome`: `Merged`, even if the resulting auto-merged version that the algorithm outputs turns out to be equal to the current or target versions